autoload/neomake.vim

```diff
@@ -524,22 +524,31 @@ function! s:command_maker_base._get_tempfilename(jobinfo) abort dict
             let s:pid = getpid()
         endif
         let slash = neomake#utils#Slash()
+
+        let dir = neomake#utils#GetSetting('tempfile_dir', self, '', a:jobinfo.ft, a:jobinfo.bufnr)
+
         let bufname = bufname(a:jobinfo.bufnr)
-        if empty(bufname)
-            let temp_file = tempname() . slash . 'neomaketmp.'.a:jobinfo.ft
-        else
-            let orig_file = neomake#utils#fnamemodify(a:jobinfo.bufnr, ':p')
-            if empty(orig_file)
+
+        let orig_file = neomake#utils#fnamemodify(a:jobinfo.bufnr, ':p')
+        if empty(dir)
+            if empty(bufname)
+                let dir = tempname()
+            elseif empty(orig_file)
```

Comment:
Now it looks like this conditional could be combined with the `if empty(bufname)`, since the body of both just set `let dir = tempname()`

Reply:
Better to have it split for coverage reporting.
It showed that `orig_file` is typically not empty if `bufname` is not at least.